File: swift/leap/Leap.swift
//
//  Leap.swift
//  Leap
//
//  Created by Taylor Smith on 9/1/15.
//  Copyright (c) 2015 TaylorSmith. All rights reserved.
//

import Foundation

struct Year {
    let calendarYear: Int
    var isLeapYear: Bool {
        return calendarYear % 4 == 0 && (calendarYear % 400 == 0 || calendarYear % 100 != 0)
    }
}

File: go/leap/leap.go
package leap

// TestVersion for exercism
const TestVersion = 1

// IsLeapYear takes an int year and returns a bool if that year is a leap year
func IsLeapYear(year int) bool {
	return year%4 == 0 && (year%400 == 0 || year%100 != 0)
}
